src/logic/safe/store/reducer/pendingTransactions.ts

```diff
@@ -46,7 +46,7 @@ export const pendingTransactionsReducer = handleActions<PendingTransactionsState
       // Omit id from the pending transactions on current chain
       const { [id]: _, ...newChainState } = state[chainId] || {}
 
-      if (Object.keys(newChainState[chainId] || {}).length === 0) {
+      if (newChainState[chainId] && Object.keys(newChainState[chainId]).length === 0) {
```

Comment:
`newChainState[chainId]` will always be undefined since `newChainState` is already `chainId` specific